### ISSUE TITLE
Consumer Groups documentation

### ISSUE THREAD

**Issue by mflendrich** [Kong/kubernetes-ingress-controller#4394]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem Statement

#3728 need to be documented

### Proposed Solution

- Document the CRD consistently with other existing CRDs
- Add a guide

### Acceptance Criteria

- [ ] The new CRD is included in the reference part of the documentation
- [ ] A guide for CGs exists

**Comment by rainest** [comment]
https://github.com/Kong/docs.konghq.com/pull/5865 starts this since AFAIK the resource code is all finalized and we're just waiting on the actual plugin binding code and gateway fixes (though we can test in DB mode without the latter).

I added a section to the plugin guide rather than creating a separate guide, since plugin binding is currently the only use of consumer groups, and a separate guide would mostly be a pared-down duplicate of the plugin guide. We hopefully can link to an anchor in the nav if we want to expose it in the ToC.
